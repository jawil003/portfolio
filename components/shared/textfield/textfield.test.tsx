import React from "react";
import { render } from "@testing-library/react";
import { Formik } from "formik";
import {
  Textfield,
  Props as TextFieldProps,
} from "./textfield";

const variants: TextFieldProps[] = [
  {
    title: "Password",
    name: "password",
  },
  {
    title: "Password",
    name: "password",
    value: "test1234",
  },
  {
    title: "Password",
    name: "password",
    value: "",
    placeholder: "test1234",
  },
];

describe("TextField", () => {
  test.each(variants)(
    "Create TextField with %p",
    (variant) => {
      const testTextField = render(
        // eslint-disable-next-line react/jsx-props-no-spreading
        <Formik
          initialValues={{
            password: "",
          }}
          onSubmit={jest.fn()}
        >
          {() => (
            <Textfield {...variant} />
          )}
        </Formik>,
      );

      expect(
        testTextField.container
          .firstChild,
      ).toMatchSnapshot();
    },
  );
});
