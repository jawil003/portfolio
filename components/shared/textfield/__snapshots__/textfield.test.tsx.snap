// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Create TextField with {"name": "password", "placeholder": "test1234", "title": "Password", "value": ""} 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: no-wrap;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  margin: calc(0.6333333333333333em / -2) calc(0px / -2);
}

.emotion-0>* {
  margin: calc(0.6333333333333333em / 2) calc(0px / 2);
}

.emotion-1 {
  font-style: normal;
  font-weight: 600;
  font-size: 1em;
}

.emotion-2 {
  border: 0.13333333333333333em solid #000000;
  box-sizing: border-box;
  border-radius: 1.25em;
  padding: 0.75em 0.8em;
  height: 3.0625em;
}

.emotion-2.emotion-2 {
  margin-bottom: 0;
}

.emotion-2.emotion-2::-webkit-input-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-2.emotion-2::-moz-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-2.emotion-2:-ms-input-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-2.emotion-2::placeholder,
.emotion-2.emotion-2::-webkit-input-placeholder,
.emotion-2.emotion-2::-ms-input-placeholder,
.emotion-2.emotion-2::-moz-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-2:focus-within {
  outline: none;
}

.emotion-3.emotion-3 {
  color: red;
  margin: 0;
  font-size: 0.75em;
  line-height: 1em;
  height: 1em;
  margin-top: 0.25em;
  margin-left: 1em;
}

<div
  class="emotion-0"
>
  <label
    class="emotion-1"
    for="password"
  >
    Password
  </label>
  <input
    class="emotion-2"
    name="password"
    placeholder="test1234"
    value=""
  />
  <div
    class="emotion-3"
  />
</div>
`;

exports[`TextField Create TextField with {"name": "password", "title": "Password", "value": "test1234"} 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: no-wrap;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  margin: calc(0.6333333333333333em / -2) calc(0px / -2);
}

.emotion-0>* {
  margin: calc(0.6333333333333333em / 2) calc(0px / 2);
}

.emotion-1 {
  font-style: normal;
  font-weight: 600;
  font-size: 1em;
}

.emotion-2 {
  border: 0.13333333333333333em solid #000000;
  box-sizing: border-box;
  border-radius: 1.25em;
  padding: 0.75em 0.8em;
  height: 3.0625em;
}

.emotion-2.emotion-2 {
  margin-bottom: 0;
}

.emotion-2.emotion-2::-webkit-input-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-2.emotion-2::-moz-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-2.emotion-2:-ms-input-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-2.emotion-2::placeholder,
.emotion-2.emotion-2::-webkit-input-placeholder,
.emotion-2.emotion-2::-ms-input-placeholder,
.emotion-2.emotion-2::-moz-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-2:focus-within {
  outline: none;
}

.emotion-3.emotion-3 {
  color: red;
  margin: 0;
  font-size: 0.75em;
  line-height: 1em;
  height: 1em;
  margin-top: 0.25em;
  margin-left: 1em;
}

<div
  class="emotion-0"
>
  <label
    class="emotion-1"
    for="password"
  >
    Password
  </label>
  <input
    class="emotion-2"
    name="password"
    value=""
  />
  <div
    class="emotion-3"
  />
</div>
`;

exports[`TextField Create TextField with {"name": "password", "title": "Password"} 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: no-wrap;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  margin: calc(0.6333333333333333em / -2) calc(0px / -2);
}

.emotion-0>* {
  margin: calc(0.6333333333333333em / 2) calc(0px / 2);
}

.emotion-1 {
  font-style: normal;
  font-weight: 600;
  font-size: 1em;
}

.emotion-2 {
  border: 0.13333333333333333em solid #000000;
  box-sizing: border-box;
  border-radius: 1.25em;
  padding: 0.75em 0.8em;
  height: 3.0625em;
}

.emotion-2.emotion-2 {
  margin-bottom: 0;
}

.emotion-2.emotion-2::-webkit-input-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-2.emotion-2::-moz-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-2.emotion-2:-ms-input-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-2.emotion-2::placeholder,
.emotion-2.emotion-2::-webkit-input-placeholder,
.emotion-2.emotion-2::-ms-input-placeholder,
.emotion-2.emotion-2::-moz-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-2:focus-within {
  outline: none;
}

.emotion-3.emotion-3 {
  color: red;
  margin: 0;
  font-size: 0.75em;
  line-height: 1em;
  height: 1em;
  margin-top: 0.25em;
  margin-left: 1em;
}

<div
  class="emotion-0"
>
  <label
    class="emotion-1"
    for="password"
  >
    Password
  </label>
  <input
    class="emotion-2"
    name="password"
    value=""
  />
  <div
    class="emotion-3"
  />
</div>
`;
